olakunlemartins8/Crowdfunding
# Implement Funding Deadline Extension

### Description

This pull request adds a new feature to the Decentralized Crowdfunding Platform that allows project creators to extend the funding deadline if needed. This can be useful in cases where a project is close to its funding goal but needs a bit more time to reach it.

### Changes

1. Added a new public function `extend-deadline` to the `crowdfunding-contract.clar` file.
2. The `extend-deadline` function takes the project title and the new deadline timestamp as parameters.
3. The function checks if the caller is the original project creator and if the current deadline has not been reached.
4. If the conditions are met, the function updates the project's deadline in the contract state.
5. Updated the `README.md` file to document the new `extend-deadline` feature.
6. Added new test cases in `crowdfunding-contract.test.js` to cover the deadline extension functionality.

### Benefits

1. Provides project creators with more flexibility to reach their funding goals by allowing them to extend the deadline if needed.
2. Enhances the overall usability and robustness of the Decentralized Crowdfunding Platform.
3. The new functionality is covered by additional test cases to ensure the stability and reliability of the platform.

### How to Test

To test the new deadline extension feature, you can follow these steps:

1. Run the existing test suite using `npx vitest` to ensure all tests pass.
2. Add a new test case in `crowdfunding-contract.test.js` that verifies the `extend-deadline` function works as expected.
3. Manually create a new project, back the project, and then try to extend the deadline using the `extend-deadline` function.
4. Verify that the project's deadline has been updated correctly.

### Checklist
- [x] Implemented the `extend-deadline` function in the Clarity contract
- [x] Updated the `README.md` file with documentation for the new feature
- [x] Added new test cases to cover the deadline extension functionality
- [x] Ensured all existing tests pass

### Reviewer Notes
N/A
